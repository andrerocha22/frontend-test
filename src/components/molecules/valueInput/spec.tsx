import React from 'react';
import { mount, shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import ValueInputContainer from './index';
import { findByTestAttr } from '../../../test/testUtils';

const defaultProps = {
  label: 'Test Label',
  icon: "iconTest",
  iconBackgroundColor: '#0f0f0f',
};

describe('test card', () => {
  const setup = (props = {}) => {
    const setupProps = { ...defaultProps, ...props };
    return shallow(<ValueInputContainer {...setupProps} />);
  };

  const setupMount = (props = {}) => {
    const setupProps = { ...defaultProps, ...props };
    return mount(<ValueInputContainer {...setupProps} />);
  };

  let component: ShallowWrapper = undefined;

  test('renders without error', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-value-input');
    expect(component.length).toBe(1);
  });

  test('card snapshot test ', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-value-input');
    expect(component).toMatchSnapshot();
  });

  test('card test props', () => {
    const wrapper = setupMount();
    expect(wrapper.prop('label')).toEqual(defaultProps.label);
    expect(wrapper.prop('icon')).toEqual(defaultProps.icon);
    expect(wrapper.prop('iconBackgroundColor')).toEqual(
      defaultProps.iconBackgroundColor
    );
  });
});
