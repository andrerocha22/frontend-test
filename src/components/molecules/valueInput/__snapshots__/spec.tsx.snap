// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test value input input for amount snapshot test  1`] = `
<styled.div
  data-test="component-value-input"
>
  <styled.p>
    Test Label
  </styled.p>
  <InputCurrency
    backgroundColor="#0f0f0f"
    icon="iconTest"
  />
</styled.div>
`;
